src/Main.elm

```diff
@@ -41,28 +150,123 @@ update msg model =
 view : Model -> Html Msg
 view model =
     div []
-        [ div [ class "book-list" ] (List.map book model.books)
+        [ div [ class "book-list" ] (List.map (\b -> bookComponent b model.cart) model.books)
+        , div [ class "cart" ] [ cartView model ]
```

Comment:
If you change the order of the args to `bookComponent`, you should be able to write the map as `List.map (bookComponent model.cart) model.books`

Reply:
I'm not quite sure I understand - do you mean if the order of args are changed in bookComponent (currently `bookComponent : Book -> Cart -> Html Msg`) the bookComponent function will be able to work with only one parameter via some sort of implication?

Reply:
Assuming that `bookComponent` has the signature `bookComponent : Cart -> Book`, then I'm pretty sure that you instead of
```elm
List.map (\book -> bookComponent model.cart book) model.books
```
You can write
```elm
List.map ( bookComponent model.cart) model.books
```
as a shorthand.

There's no need to change it, it's just FYI.